code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -0,0 +1,157 @@
+		switch(SSshuttle.emergency.mode)
+			if(SHUTTLE_ESCAPE)
+				data += list("shuttle_status" = "The station has been evacuated.")
+			if(SHUTTLE_CALL)
+				if(!SSshuttle.canRecall())
+					data += list("shuttle_status" = "The station is currently undergoing evacuation procedures.")
+
+	for(var/datum/job/prioritized_job in SSjob.prioritized_jobs)
+		if(prioritized_job.current_positions >= prioritized_job.total_positions)
+			SSjob.prioritized_jobs -= prioritized_job
+
+	for(var/datum/job_department/department as anything in SSjob.joinable_departments)
+		var/list/department_jobs = list()
+		var/list/department_data = list(
+			"name" = department.department_name,
+			"jobs" = department_jobs,
+			"color" = department.ui_color,
+			"open_slots" = 0,
+		)
+		departments += list(department_data)
+
+		for(var/datum/job/job_datum as anything in department.department_jobs)
+			var/job_availability = owner.IsJobUnavailable(job_datum.title, TRUE)
+			var/datum/outfit/outfit = job_datum.outfit
+			var/datum/id_trim/trim = initial(outfit.id_trim)
+
+			var/list/job_data = list(
+				"command" = !!(job_datum.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND),
+				"description" = job_datum.description,
+				"prioritized" = !!(job_datum in SSjob.prioritized_jobs),
+				"name" = job_datum.title,
+				"used_slots" = job_datum.current_positions,
+				"open_slots" = job_datum.total_positions < 0 ? "∞" : job_datum.total_positions,
+				"icon" = initial(trim.orbit_icon)
+			)
+
+			if(job_availability != JOB_AVAILABLE)
+				job_data["unavailable_reason"] = get_job_unavailable_error_message(job_availability, job_datum.title)
+
+			if(job_datum.total_positions < 0)
+				department_data["open_slots"] = "∞"
+
+			if(department_data["open_slots"] != "∞")
+				department_data["open_slots"] += job_datum.total_positions - job_datum.current_positions
+
+			department_jobs += list(job_data)
+
+	return data
+
+/datum/latejoin_menu/ui_status(mob/user)
+	return isnewplayer(user) ? UI_INTERACTIVE : UI_CLOSE
+
+/datum/latejoin_menu/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/latejoin_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+
+	if(!ui.user.client || ui.user.client.interviewee || !isnewplayer(ui.user))
+		return TRUE
+
+	var/mob/dead/new_player/owner = ui.user
+
+	if(action == "SelectedJob")
+		if(params["job"] == "Random")
+			var/list/dept_data = list()
+			for(var/datum/job_department/department as anything in SSjob.joinable_departments)
+				for(var/datum/job/job_datum as anything in department.department_jobs)
+					if(owner.IsJobUnavailable(job_datum.title, TRUE) != JOB_AVAILABLE)
+						continue
+					dept_data += job_datum.title
+
+			if(dept_data.len <= 0) //Congratufuckinglations
+				tgui_alert(owner, "There are literally no random jobs available for you on this server, ahelp for assistance.", "Oh No!")
+				return TRUE
+
+			var/random_job
+
+			while(random_job != JOB_CHOICE_YES)
+				var/random = pick(dept_data)
```

Comment:
Should probably do `pick_n_take`, followed by a message saying there are no jobs left to choose from.